mantidproject/mantid
# Indirect - Modify ApplyPaalmanPings to work with corrections from CalculateMonteCarloAbsorption

Currently, the ApplyPaalmanPings interface (and by extension the Apply Paalman Pings interface), will not work when supplied the output from the CalculateMonteCarloAbsorption algorithm (alternatively the Calculate Monte Carlo Absorption interface). This is as a result of four of the Paalman Pings factors being required in the ApplyPaalmanPings algorithm.

This PR adds a two-factor approximation to the ApplyPaalmanPings algorithm (used when only two correction factors are provided) and changes the name of the Apply Paalman Pings interface to Apply Absorption Corrections.

Unit Tests and documentation changes (including workflow diagram) have been included.

**To test:**
1. Navigate to Interfaces > Indirect > Corrections > Calculate Monte Carlo Absorption.
2. Provide the supplied sample and container files.
3. Set 'Sample Width' to `0.3`.
4. Set 'Sample Thickness' to `0.1`.
5. Set 'Sample Height' to `0.5`.
6. Set 'Container Front Thickness' to `0.1`.
7. Set 'Container Back Thickness' to `0.1`.
8. Enter `H2-0` as the chemical formula under 'Sample Details'
9. Enter `Al` as the chemical formula under 'Container Details'
10. Click 'Run'.
11. Navigate to the Apply Absorption Corrections tab.
12. Provide the sample file into the Input field.
13. Check the 'Use Container' check-box and provide the container file.
14. Check the 'Use Corrections' check-box, change the drop-down menu to 'Workspace' and select the created `*_Corrections` workspace group.
15. Click 'Run'.
16. Ensure the algorithm is run successfully, a corrected workspace is added to the ADS and a plot of the sample, container and corrected data are shown in the interface.

**Test Files**
[test-files.zip](https://github.com/mantidproject/mantid/files/1528245/test-files.zip)

Fixes #21316. 

**Release Notes** 

#### Changes have been added to the release notes.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
